File: core/java/src/main/java/hara/lang/protocol/IEmpty.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.iempty", name = "IEmpty")
public interface IEmpty {
  @HaraMethod(value = "empty", arity = 1)
  IEmpty empty();
}

File: core/rust/src/lang/protocol/iempty.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.iempty", name = "IEmpty")]
pub trait IEmpty {
    type Output;

    #[hara_method(value = "empty", arity = 1)]
    fn empty(&self) -> Self::Output;
}
